specs/monitoring_integration.adoc

```diff
@@ -0,0 +1,396 @@
+= Spec - Introducing a new component monitoring-integration
+
+A new component called Monitoring-Integration is introduced which integrates Grafana
+with tendrl core.
+
+
+== Problem description
+
+The new monitoring-integration component would primarily focus on dynamic
+configuration of grafana, based on the topology changes in etcd. Some of these
+responsibilities would be:
+
+* Monitoring-integration should create default dashboards in grafana using predefined json.
+  The dashboard json contains details regarding panels, metrics and alerts.
+* Configuring grafana authentication and authorization (https://github.com/Tendrl/specifications/pull/205),
+  based on changes in the primary tendrl AA scheme.
+* Keeping track of the topology changes in etcd to configure the grafana dashboards
+  to ensure that the correct nodes are part of the respective cluster dashboards.
+  (Not planned yet, for 1st milestone grafana is configured using some default
+  configuration from monitoring-integration)
+
+
+== Use Cases
+
+For 1st milestone:
+
+* Monitoring-Integration will create default dashboards in grafana.
+
+For 2nd milestone:
+
+* Grafana alerting events should be notified via tendrl.
+* Monitoring-integration should be able to configure grafana based on topology
+  changes in etcd.
+
+For 3rd milestone:
+
+* User should be able access grafana dashboard via tendrl user interface using
+  tendrl authentication.
+
+== Proposed change
+
+=== Why monitoring-integration?
+
+Grafana is used for visualizing time series data in graphs. Based on the metrics
+from datasource it will create graphs and display. Grafana supports different kinds
+of http api's to configure its environment.
+
+1. Admin
+2. Alerting
+3. Authentication
+4. Dashboard
+5. Dashboard version
+6. Data source
+7. Organization
+8. Users
+9. Preference
+10. Snapshot
+11. Others
+
+Monitoring Integration will contain json related to dashboards. Monitoring integration should
+keep track of the topology changes in etcd to configure the grafana dashboards.
+
+Monitoring integration and grafana server are installed via tendrl-ansible in server machine.
+There is just one instance of monitoring-integration and grafana installed across all clusters.
+For now when monitoring integration starts then it will create gafana dashboards
+using some predefined json files (Entire flow is not planned yet).
```

Comment:
Instead of saying this may be you can say the flow structure is out of scope of this specification.